File: haskell/p006/Main.hs
module Main where

solve :: Integer -> Integer
solve n = (s * s) - ss
  where
    s :: Integer
    s = div (n * (n + 1)) 2

    ss :: Integer
    ss = div (n * (n + 1) * ((n * 2) + 1)) 6

main :: IO ()
main = print $ solve 100

File: rust/p006/src/main.rs
fn solve(n: usize) -> usize {
    let (s, ss) = ((n * (n + 1)) / 2, (n * (n + 1) * (2 * n + 1)) / 6);
    s * s - ss
}

fn main() {
    println!("{}", solve(100));
}
